src/modules/ens/sagas.ts

```diff
@@ -0,0 +1,129 @@
+  const { land, ens } = action.payload
+  try {
+    const [eth, from] = yield getEth()
+    const nodehash = namehash(ens)
+    const ensContract = new ENS(eth, Address.fromString(ENS_ADDRESS))
+    const resolverContract = new ENSResolver(eth, Address.fromString(ENS_RESOLVER_ADDRESS))
+
+    const ipfsHash = yield call(() => ipfs.uploadRedirectionFile(land))
+    const hash = contentHash.fromIpfs(ipfsHash)
+    yield call(() => 
+      resolverContract.methods
+        .setContenthash(nodehash, `0x${hash}`)
+        .send({from})
+        .getTxHash()
+    )
+    yield call(() => 
+      ensContract.methods
+        .setResolver(nodehash, Address.fromString(ENS_RESOLVER_ADDRESS))
+        .send({from})
+        .getTxHash()
+    )
+    yield put(setENSSuccess(ens, land))
+  } catch (error) {
+    yield put(setENSFailure(ens, land, error))
+  }
+}
+
+
+function* handleGetENSRequest(action: GetENSRequestAction) {
+  const { ens, land } = action.payload
+  try {
+    const [eth] = yield getEth()
+    const nodehash = namehash(ens)
+    const ensContract = new ENS(eth, Address.fromString(ENS_ADDRESS))
+    let resolverAddress:Address = yield call(() => ensContract.methods.resolver(nodehash).call())
+
+    if (resolverAddress.toString() === ENS_EMPTY_RESOLVER) {
+      return yield put(
+        getENSSuccess(ens, { 
+          resolver: ENS_EMPTY_RESOLVER, 
+          content: ENS_EMPTY_CONTENT,
+          type: 'EmptyResolver'
+        })
+      )
+    }
+
+    const resolverContract = new ENSResolver(eth, Address.fromString(ENS_RESOLVER_ADDRESS))
+    const ipfsHash = yield call(() => ipfs.uploadRedirectionFile(land))
+    const hash = contentHash.fromIpfs(ipfsHash)
+    const currentContent = yield call(() => resolverContract.methods.contenthash(nodehash).call())
+    console.log({ens, resolver: resolverAddress.toString(), ipfsHash, hash, currentContent})
+    if (currentContent === ENS_EMPTY_CONTENT) {
+      return yield put(
+        getENSSuccess(ens, {
+          resolver: ENS_RESOLVER_ADDRESS, 
+          content: currentContent,
+          type: 'EmptyContent'
+        })
+      )
+    }
+    if (`0x${hash}` === currentContent) {
+      return yield put(
+        getENSSuccess(ens, { 
+          resolver: ENS_RESOLVER_ADDRESS, 
+          content: currentContent,
+          type: 'EqualContent'
+        })
+      )
+    }
+    yield put(
+      getENSSuccess(ens, { 
+        resolver: ENS_RESOLVER_ADDRESS,
+        content: currentContent,
+        type: 'DifferentContent'
+      })
+    )
+  } catch (error) {
+    yield put(getENSFailure(error.toString()))
+  }
+}
+
+
+function* getEth() {
+  const eth = Eth.fromCurrentProvider()
+  if (!eth) {
+    throw new Error('Wallet not found')
+  }
+
+  const fromAddress = yield select(getAddress)
+  if (!fromAddress) {
+    throw new Error(`Invalid from address: ${fromAddress}`)
+  }
+
+  const from = Address.fromString(fromAddress)
+
+  return [eth, from]
+}
```

Comment:
This is fine! but can we add a comment with a TODO that says:
`// TODO: use wallet/utils getCurrentAddress instead` ?

I've added that helper to avoid repeating code here https://github.com/decentraland/builder/blob/feat/collaborators/src/modules/wallet/utils.ts but it's not merged yet so there's no need to change this file yet

Reply:
Done! thanks for the warning!